Assets/MRTK/SDK/Experimental/Elastic/Scripts/ElasticTypes.cs

```diff
@@ -42,7 +39,77 @@ public struct ElasticExtentProperties<T>
         /// Points inside the extent to which the system will snap.
         /// </value>
         [SerializeField]
-        public T[] SnapPoints;
+        public float[] SnapPoints;
+
+        /// <value>
+        /// Distance at which snap points begin forcing the spring.
+        /// </value>
+        [SerializeField]
+        public float SnapRadius;
+    }
+
+    /// <summary>
+    /// Properties of a three-dimensional extent
+    /// in which a damped harmonic oscillator is free to move.
+    /// </summary>
+    [Serializable]
+    public struct VolumeElasticExtent
+    {
+        /// <value>
+        /// Represents the lower bound of the extent.
+        /// </value>
+        [SerializeField]
+        public Bounds StretchBounds;
+
+        /// <summary>
+        /// Whether the bounds should be respected by the system.
+        /// </summary>
+        [SerializeField]
+        public bool UseBounds;
```

Comment:
Is it valid to have UseBounds = true and repeatSnapPoints = true at the same time?

Or is it not expected and would be weird.

Reply:
Yes, that would be useful. Would be able to constrain the elastic system within a bounds, but the snap points would still "tile" within the bounds.